tests/unit/linalg/test_ode2.cpp

```diff
@@ -0,0 +1,198 @@
+         Vector du(1);
+         Vector err_u(levels);
+         Vector err_du(levels);
+         int steps = ti_steps;
+
+         t = 0.0;
+         dt = t_final/double(steps);
+         u = u0;
+         du = dudt0;
+         ode_solver->Init(*oper);
+         for (int ti = 0; ti< steps; ti++)
+         {
+            ode_solver->Step(u, du, t, dt);
+         }
+         u -= u0;
+         du -= dudt0;
+
+         err_u[0] = u.Norml2();
+         err_du[0] = du.Norml2();
+
+         std::cout<<std::setw(12)<<"Error u"
+                  <<std::setw(12)<<"Error du"
+                  <<std::setw(12)<<"Ratio u"
+                  <<std::setw(12)<<"Ratio du"
+                  <<std::setw(12)<<"Order u"
+                  <<std::setw(12)<<"Order du"<<std::endl;
+         std::cout<<std::setw(12)<<err_u[0]
+                  <<std::setw(12)<<err_du[0]<<std::endl;
+
+         for (int l = 1; l< levels; l++)
+         {
+            t = 0.0;
+            steps *=2;
+            dt = t_final/double(steps);
+            u = u0;
+            du = dudt0;
+            ode_solver->Init(*oper);
+            for (int ti = 0; ti< steps; ti++)
+            {
+               //std::cout<<t<<" "<<u(0)<<" "<< du(0)<<" ll"<<l<<std::endl;
+               ode_solver->Step(u, du, t, dt);
+               //std::cout<<t<<" "<<u(0)<<" "<< du(0)<<" ll"<<l<<std::endl;
+            }
+            u -= u0;
+            du -= dudt0;
+            err_u[l] = u.Norml2();
+            err_du[l] = du.Norml2();
+            std::cout<<std::setw(12)<<err_u[l]
+                     <<std::setw(12)<<err_du[l]
+                     <<std::setw(12)<<err_u[l-1]/err_u[l]
+                     <<std::setw(12)<<err_du[l-1]/err_du[l]
+                     <<std::setw(12)<<log(err_u[l-1]/err_u[l])/log(2)
+                     <<std::setw(12)<<log(err_du[l-1]/err_du[l])/log(2) <<std::endl;
+         }
+         delete ode_solver;
+
+         return log(err_u[levels-2]/err_u[levels-1])/log(2);
+      }
+      virtual ~CheckODE2() {delete oper;};
+   };
+   CheckODE2 check;
+
+   // Newmark-based solvers
+   SECTION("Newmark")
+   {
+      std::cout <<"\nTesting NewmarkSolver" << std::endl;
+      REQUIRE(check.order(new NewmarkSolver) + tol > 2.0 );
+   }
+
+   SECTION("LinearAcceleration")
+   {
+      std::cout <<"\nLinearAccelerationSolver" << std::endl;
+      REQUIRE(check.order(new LinearAccelerationSolver) + tol > 2.0 );
+   }
+
+   SECTION("CentralDifference")
+   {
+      std::cout <<"\nTesting CentralDifference" << std::endl;
+      REQUIRE(check.order(new CentralDifferenceSolver) + tol > 2.0 );
+   }
+
+   SECTION("FoxGoodwin")
+   {
+      std::cout <<"\nTesting FoxGoodwin" << std::endl;
+      REQUIRE(check.order(new FoxGoodwinSolver) + tol > 4.0 );
+   }
+
+   //Generalized-alpha based solvers
+   SECTION("GeneralizedAlpha")
+   {
+      std::cout <<"\nTesting GeneralizedAlpha" << std::endl;
+      REQUIRE(check.order(new GeneralizedAlpha2Solver(0.0)) + tol > 2.0 );
```

Comment:
Is it worth testing this solver for values other than `0.0`?

Reply:
Indeed will add more values.